client/scripts/main.tsx

```diff
@@ -99,6 +108,11 @@ class App extends React.Component<any, any> {
     this.setState({fishes: this.state.fishes});
   };
 
+  public updateFish = (key: string, attr: string, value: string | number) => {
+    this.state.fishes[key][attr] = value;
+    this.setState({fishes: this.state.fishes});
+  };
+
```

Comment:
Super happy with how this simple implementation to update nested fish turned out! A more robust solution would most likely take the object path to an attribute instead of individual variables. You would then filter out the attribute chain by looking at "." 's .